bug: nil pointer panic in TestBuildServerConfig when container runtime unavailable

## Bug Description

`TestBuildServerConfig` in `pkg/mcp/server/handler_test.go` panics with a nil pointer dereference when no container runtime (Docker/Podman) is available. The test uses `assert.NoError` (which continues on failure) instead of `require.NoError` (which stops the test), so when `buildServerConfig` returns `(nil, error)`, execution falls through to `tt.checkConfig(t, nil)`, causing a SIGSEGV that crashes the entire test binary and masks results from other test packages.

## Location

`pkg/mcp/server/handler_test.go:346-349`

## Reproduction

```bash
# On a machine without Docker/Podman:
go test -count=1 -v -timeout 60s ./pkg/mcp/server/...
```

## Stack Trace

```
panic: runtime error: invalid memory address or nil pointer dereference [recovered, repanicked]
[signal SIGSEGV: segmentation violation code=0x1 addr=0x30 pc=0x2c6282a]

goroutine 162 [running]:
testing.tRunner.func1.2({0x30cdba0, 0x65ab930})
testing.tRunner.func1()
panic({0x30cdba0?, 0x65ab930?})
github.com/stacklok/toolhive/pkg/mcp/server.TestBuildServerConfig.func1(...)
	pkg/mcp/server/handler_test.go:329 +0x2a
github.com/stacklok/toolhive/pkg/mcp/server.TestBuildServerConfig.func2(...)
	pkg/mcp/server/handler_test.go:349 +0x139
```

## Impact

The panic crashes the test binary, preventing all subsequent test packages from running and producing misleading `FAIL` output. The three failing subtests (`valid_config`, `custom_port`, `empty_host_defaults_to_empty`, `registry source URLs...`) all hit the same nil-dereference path.

## Suggested Fix

Replace `assert` with `require` for the error and nil checks so the test stops before calling `checkConfig` with a nil config:

```go
// handler_test.go:346-349
if tt.expectError {
    require.Error(t, err)
    require.Nil(t, runConfig)
} else {
    require.NoError(t, err)
    require.NotNil(t, runConfig)
    if tt.checkConfig != nil {
        tt.checkConfig(t, runConfig)
    }
}
```

Alternatively, add a nil guard: `if runConfig != nil && tt.checkConfig != nil`.

Additionally, these test cases should either be skipped when no container runtime is present (using `t.Skip`), or the test should mock the runtime dependency so configuration-level tests don't require Docker.

---
Found via automated test execution and stack trace analysis. Happy to submit a PR if confirmed.

--- a/pkg/mcp/server/handler_test.go
+++ b/pkg/mcp/server/handler_test.go
@@ -9,6 +9,7 @@ import (
 
 	"github.com/mark3labs/mcp-go/mcp"
 	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
 
 	regtypes "github.com/stacklok/toolhive-core/registry/types"
 	"github.com/stacklok/toolhive/pkg/runner"
@@ -340,11 +341,11 @@ func TestBuildServerConfig(t *testing.T) {
 			runConfig, err := buildServerConfig(ctx, args, tt.imageURL, tt.imageMetadata, tt.extraOpts...)
 
 			if tt.expectError {
-				assert.Error(t, err)
-				assert.Nil(t, runConfig)
+				require.Error(t, err)
+				require.Nil(t, runConfig)
 			} else {
-				assert.NoError(t, err)
-				assert.NotNil(t, runConfig)
+				require.NoError(t, err)
+				require.NotNil(t, runConfig)
 				if tt.checkConfig != nil {
 					tt.checkConfig(t, runConfig)
 				}

